src/TLuaInterpreter.cpp

```diff
@@ -10641,6 +10652,501 @@ int TLuaInterpreter::restartIrc(lua_State* L)
+        lua_error(L);
+        return 1;
+    } else {
+        fRate = lua_tonumber(L, 1);
+    }
+    
+    if (fRate > 1.0)
+        fRate = 1.0;
+    
+    if (fRate < -1.0)
+        fRate = -1.0;
+    
+    speechUnit->setRate(fRate);
+	
+	TEvent event;
+	event.mArgumentList.append(QLatin1String("ttsConfigChanged"));
+	event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+	event.mArgumentList.append(QLatin1String("rate"));
+	event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+	event.mArgumentList.append(fRate);
+	event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+	mpHost->raiseEvent(event);
+	
+	return 0;
+}
+
+
+/** ttsSetSpeechPitch( pitch )
+ *  Sets the pitch of speech playback.
+ */
+int TLuaInterpreter::ttsSetSpeechPitch(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild();
+    
+    float fPitch;
+    if (!lua_isnumber(L, 1)) {
+        lua_pushstring(L, "ttsSetSpeechPitch: wrong argument type");
+        lua_error(L);
+        return 1;
+    } else {
+        fPitch = lua_tonumber(L, 1);
+    }
+    
+    if (fPitch > 1.0)
+        fPitch = 1.0;
+    
+    if (fPitch < -1.0)
+        fPitch = -1.0;
+    
+    speechUnit->setPitch(fPitch);
+	
+	TEvent event;
+	event.mArgumentList.append(QLatin1String("ttsConfigChanged"));
+	event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+	event.mArgumentList.append(QLatin1String("pitch"));
+	event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+	event.mArgumentList.append(fPitch);
+	event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+	mpHost->raiseEvent(event);
+	
+	return 0;
+}
+
+
+/** ttsSetSpeechVolume( volume )
+ *  Sets the volume of speech playback.
+ */
+int TLuaInterpreter::ttsSetSpeechVolume(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild();
+    
+    float fVol;
+    if (!lua_isnumber(L, 1)) {
+        lua_pushstring(L, "ttsSetSpeechVolume: wrong argument type");
+        lua_error(L);
+        return 1;
+    } else {
+        fVol = lua_tonumber(L, 1);
+    }
+    
+    if (fVol > 1.0)
+        fVol = 1.0;
+    
+    if (fVol < 0.0)
+        fVol = 0.0;
+    
+    speechUnit->setVolume(fVol);
+	
+	TEvent event;
+	event.mArgumentList.append(QLatin1String("ttsConfigChanged"));
+	event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+	event.mArgumentList.append(QLatin1String("volume"));
+	event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+	event.mArgumentList.append(fVol);
+	event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+	mpHost->raiseEvent(event);
+	
+	return 0;
+}
+
+
+/** ttsGetVoices()
+ *  Lists all voices available to the current OS locale.
+ *  Currently uses the default system locale.
+ */
+int TLuaInterpreter::ttsGetVoices(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild();
+    
+    QVector<QVoice> speechVoices = speechUnit->availableVoices();
+    int i=0;
+    lua_newtable(L);
+    foreach (const QVoice &voice, speechVoices) {
+        lua_pushnumber(L, ++i);
+        lua_pushstring(L, voice.name().toLatin1().constData());
```

Comment:
We are _usually_ using Utf8 in preparation of i18n (Mudlet 4), so instead of `toLatin1()` use `toUtf8()`